src/contracts/atlas/Atlas.sol

```diff
@@ -57,10 +67,13 @@ contract Atlas is Escrow {
         // TODO: Combine this w/ call to get executionEnvironment
         DAppConfig memory dConfig = IDAppControl(userOp.control).getDAppConfig(userOp);
 
-        // Get the execution environment
-        address executionEnvironment = IAtlasFactory(FACTORY).getExecutionEnvironmentCustom(
-            userOp.from, dAppOp.control.codehash, userOp.control, dConfig.callConfig
-        );
+        // Get or create the execution environment
+        (address executionEnvironment,, bool exists) =
+            IAtlasFactory(FACTORY).getExecutionEnvironment(userOp.from, userOp.control);
+
+        if (!exists) {
+            executionEnvironment = createExecutionEnvironment(userOp.from, userOp.control);
```

Comment:
Can we combine these into one function at the factory level so that we don't have to make two external calls?

Reply:
Corrected in https://github.com/FastLane-Labs/atlas/pull/61/commits/cfa28c0b8ed12f1898dafd9f60dadd315f938e80